nominal/nominal.py

```diff
@@ -1,628 +1,316 @@
 from __future__ import annotations
 
-import copy
-import io
-import os
-from datetime import datetime, timedelta
-from math import floor
+import shutil
+from datetime import datetime
 from pathlib import Path
-from typing import Mapping, Optional, Sequence
-
-import jsondiff as jd
-import keyring as kr
-import polars as pl
-import requests
-from dateutil import parser
-from jsondiff import diff
-from rich import print
-
-from ._api.combined.ingest_api import (
-    AbsoluteTimestamp,
-    IngestService,
-    IngestSource,
-    RelativeTimestamp,
-    S3IngestSource,
-    TimestampMetadata,
-    Iso8601Timestamp,
-    TimestampType,
-    TriggerIngest,
+from threading import Thread
+from typing import TYPE_CHECKING, BinaryIO, Mapping, Sequence
+
+import dateutil.parser
+
+from .exceptions import NominalError
+from .sdk import Attachment, Dataset, NominalClient, Run
+from ._utils import (
+    FileType,
+    FileTypes,
+    IntegralNanosecondsUTC,
+    TimestampColumnType,
+    _datetime_to_integral_nanoseconds,
+    reader_writer,
 )
-from ._utils import PayloadFactory, create_service, default_filename
-from .cloud import ENDPOINTS, _auth_help_blurb, get_app_base_url, get_base_url
 
+if TYPE_CHECKING:
+    import pandas as pd
+    import polars as pl
 
-class Dataset(pl.DataFrame):
-    """
-    Dataset inherits from Polars DataFrame for its rich display, ingestion, and wrangling capabilities.
-
-    Parameters
-    ----------
-    data : various, optional
-        The input data for the dataset. This can be in any format supported by Polars DataFrame.
-    filename : str, optional
-        The name of the dataset file. Default is None.
-    overwrite : bool, optional
-        A flag to indicate whether to overwrite an existing file during upload. Default is False.
-    properties : dict, optional
-        A dictionary of additional properties associated with the dataset. Default is an empty dictionary.
-    description : str, optional
-        A brief description of the dataset. Default is an empty string.
-
-    Attributes
-    ----------
-    s3_path : str or None
-        The S3 path where the dataset is stored after upload. Initially None.
-    filename : str
-        The name of the dataset file.
-    properties : dict
-        A dictionary of additional properties associated with the dataset.
-    description : str
-        A brief description of the dataset.
-    rid : str or None
-        The dataset's RID (Resource ID) after registration on the Nominal platform. Initially None.
-    dataset_link : str
-        A URL link to the dataset on the Nominal platform. Initially an empty string.
-
-    Methods
-    -------
-    upload(overwrite=False)
-        Uploads and registers the dataset on the Nominal platform.
+
+_default_connection: NominalClient | None = None
+
+
+def set_default_connection(base_url: str, token: str) -> None:
```

Comment:
should I create separate functions for setting the base URL and token? should I do that in addition to, or instead of, set_default_connection()?

Reply:
in examples, it's a bit easier to copy/paste and edit one line at a time when they are separate functions. if you copy/paste a multiline thing into a repl, you can't really go back to edit it

Reply:
I quite liked the `set_token` functionality of the previous api, and nice to have `keyring` secure it and retrieve it so it doesn't have to be manually included in every script. I think setting tokens should be separate from setting URIs (every customer has to set their token - most customers will not need to override the api URI)

my personal opinion is that the python client should assume the public API url, and only require configuration for on-prem uris (which we'll happen to use too for development)

Reply:
> I quite liked the set_token functionality of the previous api, and nice to have keyring secure it and retrieve it so it doesn't have to be manually included in every script.

yeah agreed, was balancing this need vs relying on mutable system state which I don't really like. Biasing against someone running a script that "works on their machine" and then a coworker runs the same script and it doesn't work.

Reply:
counterpoint: biasing against folks checking in tokens

Reply:
added keyring functionality back in